scripts/mk_all.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Damiano Testa
+-/
+import Cli.Basic
+import Lean.Util.Path
+import Lake.CLI.Main
+
+/-!
+# Script to create a file importing all files from a folder
+
+This file declares a command to gather all Lean files from a folder into a single Lean file.
+-/
+
+open Lean System.FilePath
+
+/-- `getAll git ml` takes all `.lean` files in the dir `ml` (recursing into sub-dirs) and
+returns the string
+```
+import file₁
+...
+import fileₙ
+```
+The input `git` is a `Bool`ean flag:
+* `true` means that the command uses `git ls-files` to find the relevant files;
+* `false` means that the command recursively scans all dirs searching for `.lean` files.
+-/
+def getAll (git : Bool) (ml : String) : IO String := do
```

Comment:
Would it be worth to split this up into a function that returns a `List String` of the form `["Mathlib.Algebra.Algebra.Basic", "Mathlib.Algebra.Algebra.Defs", ...]`? I can imagine that form of output is more useful for other scripts.

Reply:
Case in point: I need this functionality in #13245. (An array of strings is also fine for me.)

To share it, presumably this ought to go in a better place than this script (in `Mathlib/Util` perhaps? or in `Lake`).
In any case, moving this to a better place can also happen in a follow-up.

Reply:
Thanks, I like this suggestions and I implemented it!